### PR TITLE
Enable goto-analyzer-simplify regression tests

### PR DESCRIPTION
**Pull request by tautschnig**
Updated scripts and test specifications to work in current environments.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: regression/goto-analyzer-simplify/Makefile [hunk 1]
```diff
@@ -1,22 +1,15 @@
...
-		exit 1; \
-	fi
+tests.log: ../test.pl
+	@../test.pl -e -p -c "../chain.sh ../../../src/goto-analyzer/goto-analyzer"
 
 show:
 	@for dir in *; do \
 		if [ -d "$$dir" ]; then \
-			vim -o "$$dir/*.c" "$$dir/*.out"; \
+			vim -o "$$dir/*.java" "$$dir/*.out"; \
```
Comment (martin-cs) [inline regression/goto-analyzer-simplify/Makefile:12]: ??? Why change this?

**Review by martin-cs — APPROVED** [review]
Changes look good and preserve the original configuration.  I have also checked the tests and I think the results are correct.

File: regression/goto-analyzer-simplify/Makefile (hunk 1 above)
Comment (tautschnig) [reply regression/goto-analyzer-simplify/Makefile:12]: Oh, my bad, I copied `regression/goto-analyzer/Makefile`. I suppose both should be changed, or perhaps we should just get rid of this `show` target. Will create a PR for this.

Comment (tautschnig) [reply regression/goto-analyzer-simplify/Makefile:12]: Finally done in https://github.com/diffblue/cbmc/pull/5957.

**Pull request merged**